ruby_programming/advanced_ruby/blocks.md

```diff
@@ -475,24 +475,24 @@ After coming to grips with the information in this lesson you'll be a block, pro
 </div>
 
 ### Knowledge Check
- 
+
 This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
- 
-- What is a block?
-- How is a block like a method?
-- How is a block different from a method?
-- What are the two ways to declare a block?
-- How do you return data from a block?
-- How can your methods collect the return data from a block?
-- What happens if you include a `return` statement in a block?
-- Why would you use a block instead of just creating a method?
-- What does `yield` do?
-- How do you pass arguments to a block from within a method?
-- How do you check whether a block was actually passed in?
-- What is a proc?
-- What is a lambda?
-- What's the difference between a proc and a block?
-- When would you use a proc instead of a block?
-- What's different between a lambda and a proc?
-- How do you convert a proc to a block?
-- How do you convert a block to a proc?
+
+- <a class="knowledge-check-link" href="https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/#Understanding_Ruby_Blocks">What is a block?</a>
+- <a class="knowledge-check-link" href="https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/#Understanding_Ruby_Blocks">How is a block like a method?</a>
+- <a class="knowledge-check-link" href="#block-basics">How is a block different from a method?</a>
+- <a class="knowledge-check-link" href="#block-basics">What are the two ways to declare a block?</a>
+- <a class="knowledge-check-link" href="#return-value">How do you return data from a block?</a>
+- <a class="knowledge-check-link" href="#collect-return">How can your methods collect the return data from a block?</a>
+- <a class="knowledge-check-link" href="#explicit-return">What happens if you include a `return` statement in a block?</a>
+- <a class="knowledge-check-link" href="#lambdas">Why would you use a block instead of just creating a method?</a>
```

Comment:
I'm also not understanding quite how this question is answered in the linked section, can you point that out for me?

Reply:
I didn't find a good answer for the question online 488 in the lesson materials. The linked section describes passing the same block to multiple methods to keep from typing out the same code over and over again. That would be a reason to use a block instead of a method.

Reply:
I don't think that's a great match. I think this is a question that should probably just be removed as well.